### PR TITLE
unit test: Added Uniswap V3 unit test

### PR DESCRIPTION
**Pull request by Turupawn**
Replaced the previous challenge with a foundry testing on Scroll Sepolia.

### CODE REVIEW

File: test/UniV3Swap.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "../src/UniV3Swap.sol";
+
+contract UniV3SwapTest is Test {
+    UniV3Swap public uniV3Swap;
+    address SCROLL_SEPOLIA_UNIV3_ROUTER_V2 = 0x17AFD0263D6909Ba1F9a8EAC697f76532365Fb95;
```
Comment (emilyJLin95) [inline test/UniV3Swap.sol:9]: Would devs be able to fork anvil to match Scroll Sepolia?

Comment (emilyJLin95) [reply test/UniV3Swap.sol:9]: Or that only that they'll test directly on Scroll Sepolia testnet

Comment (emilyJLin95) [reply test/UniV3Swap.sol:9]: I guess for automated checking though, once Isabelle implements, this wouldn't be doable unless we had a in-browser instance of a forked Scroll Sepolia?

File: README.md [hunk 2]
```diff
@@ -1,71 +1,11 @@
-# 🚀 Master the Simple Token Swap Challenge
+# Swap Token Test
 
-Congratulations for completing ERC20 Basic Challenge!
+Simple Token Swap challenge for the LevelUp platform. Complete the `swap` function at `src/UniV3Swap.sol` and run the test to check the correctness of your answer.
 
-You have now grasped the functionality of ERC20 tokens. Ever wondered how you can use these tokens to swap to another token (i.e WETH to USDC)?
+## Running the test
 
-Let's learn how the `exactInputSingle()` function in Uniswap works! 💪
```
Comment (emilyJLin95) [inline README.md:7]: Why remove the instructions here?

Comment (Turupawn) [reply README.md:7]: I removed it to match the lending challenge:
https://github.com/LevelUpWeb3/SimpleLending-Challenge?tab=readme-ov-file#aave-lender-demo

Should we use the previous format in both?

File: test/UniV3Swap.sol (hunk 1 above)
Comment (Turupawn) [reply test/UniV3Swap.sol:9]: > Would devs be able to fork anvil to match Scroll Sepolia?

Yes, they are expected to run the test on sepolia like this: `forge test --fork-url https://scroll-testnet-public.unifra.io`

 

> I guess for automated checking though, once Isabelle implements, this wouldn't be doable unless we had a in-browser instance of a forked Scroll Sepolia?

The two options are implementing mock contracts (not ideal but can work) or having a backend that runs the tests. In the last call Isabelle mentioned that she was working on a backend but I'm not sure if I understood that correctly

Comment (emilyJLin95) [reply test/UniV3Swap.sol:9]: As per our slack convo, let's update the README to include:

Short description
Link to level up
Instructions to running the challenge

**Review by emilyJLin95 — APPROVED** [review]
LGTM

**Comment by emilyJLin95** [comment]
@Turupawn @PrintRH can you merge

**Comment by PrintRH** [comment]
@emilyJLin95 Merged

**Pull request merged**